cmd/eksctl-anywhere/cmd/validate.go

```diff
@@ -0,0 +1,166 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/providers/cloudstack/decoder"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type validateOptions struct {
+	clusterOptions
+	skipIpCheck           bool
+	hardwareCSVPath       string
+	tinkerbellBootstrapIP string
+}
+
+var valOpt = &validateOptions{}
+
+var validateCmd = &cobra.Command{
+	Use:          "validate",
+	Short:        "Validate configuration",
+	Long:         "This command is used to validate eksctl anywhere configurations",
+	PreRunE:      preRunValidate,
+	SilenceUsage: true,
+	RunE:         valOpt.validateCluster,
+}
+
+func init() {
+	expCmd.AddCommand(validateCmd)
+	validateCmd.Flags().StringVarP(&valOpt.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	validateCmd.Flags().StringVarP(
+		&valOpt.hardwareCSVPath,
+		TinkerbellHardwareCSVFlagName,
+		TinkerbellHardwareCSVFlagAlias,
+		"",
+		TinkerbellHardwareCSVFlagDescription,
+	)
+	validateCmd.Flags().StringVar(&valOpt.tinkerbellBootstrapIP, "tinkerbell-bootstrap-ip", "", "Override the local tinkerbell IP in the bootstrap cluster")
+	validateCmd.Flags().BoolVar(&valOpt.skipIpCheck, "skip-ip-check", false, "Skip check for whether cluster control plane ip is in use")
+	validateCmd.Flags().StringVar(&valOpt.bundlesOverride, "bundles-override", "", "Override default Bundles manifest (not recommended)")
+	validateCmd.Flags().StringVar(&valOpt.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
+
+	if err := validateCmd.MarkFlagRequired("filename"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+}
+
+func preRunValidate(cmd *cobra.Command, args []string) error {
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		err := viper.BindPFlag(flag.Name, flag)
+		if err != nil {
+			log.Fatalf("Error initializing flags: %v", err)
+		}
+	})
+	return nil
```

Comment:
I know this is copied but since this is new code lets do some simple things better.

```suggestion
    var err error
	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
		if err != nil {
		  return
		}
		err = viper.BindPFlag(flag.Name, flag)
	})
	return err
```